File: 100-Days-Series/Day 68/HourglassPatternWithNumber.cpp
#include<iostream>
using namespace std;
// Function to print Hourglass Pattern With Number
void getPrintPattern(int n){
    for (int i = n; i >= 1; i--) {
            for (int j = 0; j < n - i; j++) {
                cout<<"  ";
            }
            for (int j = 1; j <= 2 * i - 1; j++) {
                cout<<j<<" ";
            }
            cout<<endl;
        }
        for (int i = 2; i <= n; i++) {
            for (int j = 0; j < n - i; j++) {
                cout<<"  ";
            }
            for (int j = 1; j <= 2 * i - 1; j++) {
                cout<<j<<" ";
            }
            cout<<endl;
        }
}
// Let's Start Here
int  main(){
    int rows;
    cout<<"\nEnter the number of rows : ";
    cin>>rows;
    getPrintPattern(rows);
    return 0;
}

File: 100-Days-Series/Day 68/HourglassPatternWithNumber.js
// Function to print Hourglass Pattern with number.
function getPrintPattern(n){
    for (let i = n; i >= 1; i--) {
        line="";
        for (let j = 0; j < n - i; j++) {
            line+="  ";
        }
        for (let j = 1; j <= 2 * i - 1; j++) {
            line+=j+" ";
        }
        console.log(line);
    }
    for (let i = 2; i <= n; i++) {
        line="";
        for (let j = 0; j < n - i; j++) {
            line+="  ";
        }
        for (let j = 1; j <= 2 * i - 1; j++) {
            line+=j+" ";
        }
        console.log(line);
    }
}

// Let's Start Here
rows = 5;
getPrintPattern(rows);